### PR TITLE
feat: filters for stops

### PR DESCRIPTION
**Pull request by alainajaffer**

### CODE REVIEW

File: packages/api/client.ts [hunk 1]
```diff
@@ -372,6 +377,9 @@ export const getServiceStops = async (dbClient: Kysely<Database>, input: Service
         .where("fromStop.stopType", "not in", ignoredStopTypes)
         .where("toStop.stopType", "not in", ignoredStopTypes)
         .where((qb) => qb.where("fromStop.status", "=", "active").orWhere("toStop.status", "=", "active"))
+        .$if(!!input.modes?.[0], (qb) => qb.where("mode", "in", input.modes ?? ["---"]))
```
Comment (alvak1) [inline packages/api/client.ts]: should be `services.mode` instead of `mode` and use the fully qualifying name

File: packages/api/client.ts [hunk 2]
```diff
@@ -372,6 +377,9 @@ export const getServiceStops = async (dbClient: Kysely<Database>, input: Service
...
+        .$if(!!input.stopTypes?.[0], (qb) => qb.where("stopType", "in", input.stopTypes ?? ["---"]))
+        .$if(!!input.busStopType, (qb) => qb.where("busStopType", "=", input.busStopType ?? "---"))
```
Comment (alvak1) [inline packages/api/client.ts:382]: use fully qualifying column name

Comment (alvak1) [inline packages/api/client.ts]: use fully qualifying column name

File: packages/api/get-service-stops.ts [hunk 3]
```diff
@@ -19,8 +19,29 @@ export const getQueryInput = (event: APIGatewayEvent): ServiceStopsQueryInput =>
...
+
+    if (filteredModesArray.length !== modesArray.length) {
+        throw new ClientError("Invalid mode provided");
+    }
+
     return {
         serviceId: Number(serviceId),
+        ...(pathParameters?.busStopType ? { busStopType: pathParameters.busStopType } : {}),
+        ...(filteredModesArray && filteredModesArray.length > 0 ? { modes: filteredModesArray } : {}),
+        ...(stopTypesArray && stopTypesArray.length > 0 ? { stopTypes: stopTypesArray } : {}),
```
Comment (alvak1) [inline packages/api/get-service-stops.ts:54]: `busStopType` is not read from pathParameters and passed as input to client

Comment (alainajaffer) [reply packages/api/get-service-stops.ts:54]: if is line 42

File: packages/api/client.ts (hunk 1 above)
Comment (alainajaffer) [reply packages/api/client.ts]: done

File: packages/api/client.ts (hunk 2 above)
Comment (alainajaffer) [reply packages/api/client.ts]: done

**Review by alvak1 — COMMENTED** [review]
Approved

**Review by alvak1 — APPROVED** [review]
Approved

**Pull request merged**